connectors/api/src/main/java/com/datastax/oss/dsbulk/connectors/api/DefaultErrorRecord.java

```diff
@@ -15,56 +15,40 @@
  */
 package com.datastax.oss.dsbulk.connectors.api;
 
-import com.datastax.oss.driver.shaded.guava.common.base.Suppliers;
 import com.datastax.oss.driver.shaded.guava.common.collect.ImmutableList;
 import com.datastax.oss.driver.shaded.guava.common.collect.ImmutableSet;
 import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
 import java.net.URI;
 import java.util.Collection;
 import java.util.Set;
-import java.util.function.Supplier;
 
 /** A record that could not be fully parsed from its source. */
 public class DefaultErrorRecord implements ErrorRecord {
 
-  private final @NonNull Object source;
-  private final @NonNull Supplier<URI> resource;
+  private final @Nullable Object source;
+  private final @NonNull URI resource;
   private final long position;
   private final @NonNull Throwable error;
 
   /**
    * Creates a new error record.
    *
-   * @param source the record's source.
+   * @param source the record's source; may be null if the source cannot be determined or should not
+   *     be retained.
    * @param resource the record's resource.
    * @param position the record's position.
    * @param error the error.
    */
   public DefaultErrorRecord(
-      @NonNull Object source, @NonNull URI resource, long position, @NonNull Throwable error) {
-    this(source, () -> resource, position, error);
-  }
-
-  /**
-   * Creates a new error record.
-   *
-   * @param source the record's source.
-   * @param resource the record's resource; will be memoized.
-   * @param position the record's position.
-   * @param error the error.
-   */
-  public DefaultErrorRecord(
-      @NonNull Object source,
-      @NonNull Supplier<URI> resource,
-      long position,
-      @NonNull Throwable error) {
+      @Nullable Object source, @NonNull URI resource, long position, @NonNull Throwable error) {
     this.source = source;
-    this.resource = Suppliers.memoize(resource::get);
```

Comment:
why don't we need this memoize anymore?

Reply:
All of this came from the assumption that row records (records produced by mapping a database row – for UNLOAD operations) had URI resources that were expensive to calculate because they involved string concatenation and inspection of `ExecutionInfo` and `ColumnDefinitions` objects.

And URI resources for row records are, in fact, not very useful (URI resources are especially useful for records produced from files since the URI resource in this case _is_ the file path).

So I just decided to simplify row URI resources to the minimum: they are now simply of the form `cql://keyspace/table`, and that can be calculated once for all rows.

As a consequence, the memoization is not necessary anymore.